crates/sui-core/src/authority/authority_per_epoch_store.rs

```diff
@@ -170,15 +170,24 @@ impl AuthorityPerEpochStore {
         let wal_path = parent_path.join("recovery_log");
         let wal = Arc::new(DBWriteAheadLog::new(wal_path));
         let epoch_alive = NotifyOnce::new();
-        Self {
+
+        let this = Self {
             committee,
             tables,
             reconfig_state_mem: RwLock::new(reconfig_state),
             epoch_alive,
             consensus_notify_read: NotifyRead::new(),
             end_of_publish: Mutex::new(end_of_publish),
             wal,
+        };
+
+        if this.get_last_checkpoint_boundary().is_none() {
+            debug!("Inserting initial checkpoint boundary into empty table");
+            this.insert_checkpoint_boundary(0, 0)
```

Comment:
Would this cause we always miss tx at hight 0? (since the second 0 will become from_height_excluded)

Reply:
I think this should be ok. There are two integers here - pending checkpoint height and commit boundary.
Pending checkpoint height is read by checkpoint builder only, and it's ok if we start creating them from 1.
Setting commit boundary to 0 here means that, we will miss transactions committed with leader round 0.

I *think* (and I can confirm it by looking at logs [here](https://mysten.grafana.net/goto/MA_t-U5Vk?orgId=1) for example) that narwhal can't make a commit with leader certificate round=0(e.g. first leader round is 2), so we are good in that case.

Alternatively, we could modify the code below and make `from_height_excluded` an option in `last_checkpoint` function and handle it correctly. But I think either way works

Reply:
It's not that hard to fix this though. There is not much good reason to leave this potential issue there.

Reply:
Fair enough, although the question was whether it actually is an issue.

I'll change `last_checkpoint` - does anyone care whether I leave the other code in place so the table always starts at (0,0)? Probably doesn't matter, but I slightly prefer having the table start from a known state (as opposed to depending on when the first commit happens).

Reply:
https://github.com/MystenLabs/sui/blob/1dacc0fd988bb58b36781b54426b28bbe9a0cd66/narwhal/consensus/src/bullshark.rs#L94
Leader round cannot be < 2.

https://github.com/MystenLabs/sui/blob/1dacc0fd988bb58b36781b54426b28bbe9a0cd66/narwhal/consensus/src/utils.rs#L23
A sub-dag leader cannot be at round 0, because `last_committed_round` is initialized to 0 and each sub-dag leader's min round is `last_committed_round + 2`